### PR TITLE
[#153119879] Upgrade Terraform

### PR DESCRIPTION
**Pull request by 46bit**
## What

Upgrade Terraform to the latest stable version.

## How to review

1. Review the individual commit messages in order.
1. Confirm that Travis tests have passed.
1. Run the pipelines against an existing environment and confirm that all Terraform jobs are noops.
1. Run the pipelines against a new environment and confirm that it provisions OK.
1. Manually run the CDN instance Terraform (`make … setup_cdn_instances ACTION=…`)
1. Manually run the [Upload Certs Terraform](https://github.com/alphagov/paas-cf/blob/master/doc/ssl_certs.md) (can be tested with https://github.com/alphagov/paas-credentials/commit/2a62521fc9ac033de57d99828328517b3a5fc4fd)
1. Manually run the [Pingdom Terraform](https://github.com/alphagov/paas-cf/blob/master/doc/pingdom.md) with alphagov/paas-credentials#103

(we have already performed some variations of the testing so happy to talk about whether this can be omitted if it proves too time consuming)

:bee: :bee: :bee: **Bee-fore merging:** :bee: :bee: :bee:

1. Merge alphagov/paas-credentials#103
1. Merge alphagov/paas-docker-cloudfoundry-tools#107
1. Wait for Docker Hub to build and tag the container.
1. Amend the `wip` commit to use the Docker image tag that matches the merge commit in alphagov/paas-docker-cloudfoundry-tools.

:bee: :bee: :bee: **After bee-ploying:** :bee: :bee: :bee:

1. Manually run the CDN instance Terraform (`make … setup_cdn_instances ACTION=…`)
1. Manually run the [Upload Certs Terraform](https://github.com/alphagov/paas-cf/blob/master/doc/ssl_certs.md)
1. Manually run the [Pingdom Terraform](https://github.com/alphagov/paas-cf/blob/master/doc/pingdom.md)

## Who can review

Not @dcarley or @46bit

### CODE REVIEW

**Comment by samcrang** [comment]
I just tried to apply this to a new environment and it hung with:

```
Do you want to perform these actions?
	
  Terraform will perform the actions described above.
	
  Only 'yes' will be accepted to approve.
	

	
  Enter a value: 
```

File: concourse/pipelines/create-cloudfoundry.yml [hunk 1]
```diff
@@ -969,9 +972,13 @@ jobs:
                 mkdir generated-certificates
                 tar xzf cf-certs/cf-certs.tar.gz -C generated-certificates
 
-                terraform apply -var-file="paas-cf/terraform/((aws_account)).tfvars" \
+                cp cf-tfstate/cf.tfstate updated-tfstate/cf.tfstate
+                terraform init paas-cf/terraform/cloudfoundry
+                terraform apply \
+                  -var-file="paas-cf/terraform/((aws_account)).tfvars" \
```
Comment (dcarley) [inline concourse/pipelines/create-cloudfoundry.yml]: @samcrang It looks like we missed `--auto-approve=true` here. I might have broken it while rebasing. I can add it now if you're alright with me rebasing the branch?

Comment (samcrang) [reply concourse/pipelines/create-cloudfoundry.yml]: I'm okay with rebasing--rebasing is my favourite thing ever.

Comment (dcarley) [reply concourse/pipelines/create-cloudfoundry.yml]: Then have I got a force push for you! Fixed now.

**Comment by samcrang** [comment]
I've updated the Docker image tag.

**Pull request merged**